### PR TITLE
MCLOUD-5511: Add Multi-website Support into Cloud Docker

### PR DESCRIPTION
**Pull request by mohanelamurugan**
### Description
Added multi website support in cloud docker(Appended magneto-vars.php to php.ini) .

https://jira.corp.magento.com/browse/MCLOUD-5511: 
1. Start the magento cloud docker, set to developer deploy mode following this tutorial: https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-config.html
2. Set up a new website, store, store view and use this tutorial to set up a new website: https://devdocs.magento.com/guides/v2.3/cloud/project/project-multi-sites.html
3. Assuming we plan to have http://second.magento2.docker/ for the new website that we are adding, Add second.magento2.docker to hosts file.
4. Visit http://second.magento2.docker/

### Fixed Issues (if relevant)

[MCD-104](https://github.com/magento/magento-cloud-docker/issues/104)

### Manual testing scenarios

1. Add both websites base url to hosts
2. Visit each website's URL from browser
3. Secondary website URL should redirect to the secondary website instead of redirecting to the base website's URL.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: images/php/7.1-cli/etc/php-cli.ini [hunk 1]
```diff
@@ -7,3 +7,4 @@ always_populate_raw_post_data = -1
 max_input_vars = 10000
 session.gc_probability = 1
 opcache.enable = 1
+auto_append_file = /app/magento-vars.php
```
Comment (shiftedreality) [inline images/php/7.1-cli/etc/php-cli.ini]: Please add new line

File: images/php/fpm/etc/php-fpm.ini [hunk 2]
```diff
@@ -1,2 +1,3 @@
 memory_limit = !PHP_MEMORY_LIMIT! ; Variable: PHP_MEMORY_LIMIT
 opcache.enable = 1
+auto_append_file = /app/magento-vars.php
```
Comment (shiftedreality) [inline images/php/fpm/etc/php-fpm.ini]: Please add new line

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: images/php/7.1-cli/etc/php-cli.ini (hunk 1 above)
Comment (mohanelamurugan) [reply images/php/7.1-cli/etc/php-cli.ini]: Added. Pls review again

File: images/php/fpm/etc/php-fpm.ini [hunk 3]
```diff
@@ -1,2 +1,4 @@
+ff
```
Comment (NadiyaS) [inline images/php/fpm/etc/php-fpm.ini]: Please remove this

File: images/php/cli/etc/php-cli.ini [hunk 4]
```diff
@@ -1,3 +1,4 @@
+git stataus
```
Comment (NadiyaS) [inline images/php/cli/etc/php-cli.ini]: please remove this line

**Review by shiftedreality — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
Created internal task MAGECLOUD-5584

**Comment by shiftedreality** [comment]
To run in Production mode:

1. Create file `docker-compose.override.yaml`:

```yaml
version: '2.1'
services:
  web:
    volumes:
      - './magento-vars.php:/app/magento-vars.php:ro'
```

**Comment by andriyShevtsov** [comment]
@shiftedreality Please take a look. With production mode and  `docker-compose.override.yaml` I have error  
`Fatal error: Cannot redeclare isHttpHost() (previously declared in /app/magento-vars.php:20) in /app/magento-vars.php on line 20`

**Comment by shiftedreality** [comment]
@andriyShevtsov it looks like the trick with `docker-compose.override.yml` is not needed. This issue somehow relates to Docker itself and can be avoided by simple adjustment:

```php
if (!function_exists('isHttpHost')) {
    function isHttpHost($host)
    {
        if (!isset($_SERVER['HTTP_HOST'])) {
            return false;
        }
        return strpos(str_replace('---', '.', $_SERVER['HTTP_HOST']), $host) === 0;
    }
}
```

**Comment by shiftedreality** [comment]
I just realized it will break the default behavior. 

@mohanelamurugan could you please check this PR in Production mode if you have the same problem?

**Comment by mohanelamurugan** [comment]
Sure @shiftedreality , Let me check this

**Comment by mohanelamurugan** [comment]
@shiftedreality 

I didn't use the docker-compose.override.yml
I tested it in production mode. I was able to run deploys, post deploys, cache clean, index:reindex as well.
Also multi site redirections are also working fine.
The first one runs in the URL magento2.docker whereas the other one is second.magento2.docker.
Also from the attached image below, You can find that the setup is running in production mode.
PFA 
Screen Shot 2020-03-19 at 4 45 47 PM
Screen Shot 2020-03-19 at 4 45 53 PM
Screen Shot 2020-03-19 at 5 18 08 PM

**Comment by andriyShevtsov** [comment]
Thank you. Most likely that issue was not related to your PR. QA approved.

**Pull request merged**